packages/@lwc/state/src/contextful-lwc.ts

```diff
@@ -59,6 +59,14 @@ export class ContextfulLightningElement extends LightningElement {
             }
           });
         }
+
+        if (providedContextVarieties.has(contextVariety)) {
+          console.error(
+            'Multiple contexts of the same variety were provided. Only the first context will be used.',
+          );
```

Comment:
This looks good. Only possible addition: we might want to log _once_. Something like this:

```javascript
let multipleContextWarningShown = false;

/// ...

if (providedContextVarieties.has(contextVariety)) {
  if (!multipleContextWarningShown){
    multipleContextWarningShown = true;
    console.error(
      'Multiple contexts of the same variety were provided. Only the first context will be used.',
    );
  }
}
```

We use patterns like this in the LWC codebase. That said, I'm hoping that this is a rare occurrence. So let's see if it becomes a problem, and return to this if necessary.